src/validation_workflow/validation_args.py

```diff
@@ -114,7 +114,7 @@ def __init__(self) -> None:
         parser.add_argument(
             "-f",
             "--force-https",
-            action="store_true",
+            action="store_false",
```

Comment:
Why this change?

Reply:
Bit confused on this parameter actually. This is always true with current implementation. Now , It is true by default, when -----force-https is passed it will store false.

To remove the confusion we can change the name to --do-not force-https or something similar and assign a value False by default, if provided we check for security plugin and run http/https as required.

Reply:
Why do we keep doing negative params? I still dont understand your point here. If --force-https, then it will always check https, if not / default it will check https or http based on security plugin presence, isnt it?

Reply:
I think we want to make the validation workflow confined to `https` all the time. Unless an external argument  --force-https is provided we do not want to validate without security plugin(i.e http)
Adding @gaiksaya

Reply:
Right! Security is always assumed to be enabled by default especially for validation.

Reply:
Hi @peterzhuamazon, addressed all the changes except this. Can I know your thoughts?

Reply:
After talking to @Divyaasm as there are some confusion.

The param was meant to `disable` https-only check, aka allow http if security plugin not present.
See this flow:

![validation force-https drawio](https://github.com/opensearch-project/opensearch-build/assets/65193828/e543f336-be63-4e4c-8a20-afb4468df8b9)

Therefore, the initial naming of `--force-https` is not applicable anymore.

Suggest using `--allow-http` so that, if such parameter not present it will always force to check https no matter what; else, if the parameter present it will allow for `http` check if security plugin is not present.

Thanks.

Reply:
Resolved now.